nvflare/app_opt/tracking/mlflow/mlflow_receiver.py

```diff
@@ -0,0 +1,324 @@
+            else:
+                experiment_id = experiment.experiment_id
+
+            self.logger.info(f"Experiment={experiment}")
+        return experiment_id
+
+    def save(self, fl_ctx: FLContext, shareable: Shareable, record_origin: str):
+        if self.time_start == 0:
+            self.time_start = timeit.default_timer()
+
+        dxo = from_shareable(shareable)
+        data = AnalyticsData.from_dxo(dxo, receiver=LogWriterName.MLFLOW)
+        if not data:
+            return
+
+        # mlflow_client = self.get_mlflow_client(record_origin)
+        # run_id = self.get_run_id(record_origin)
+
+        if data.data_type == AnalyticsDataType.TEXT:
+            # not currently supported
+            pass
+        elif data.data_type == AnalyticsDataType.MODEL:
+            # not currently supported
+            pass
+        elif data.data_type == AnalyticsDataType.IMAGE:
+            # not currently supported
+            pass
+        else:
+            self.buffer_data(data, record_origin)
+            self.flush_buffer(record_origin)
+
+    def buffer_data(self, data: AnalyticsData, record_origin: str) -> None:
+        """Buffer the data to send later.
+
+        A buffer for each data_type is in each site_buffer, all of which are in self.buffer
+
+        Args:
+            data (AnalyticsData): Data.
+            record_origin (str): Origin of the data, or site name.
+        """
+        site_buffer = self.buffer[record_origin]
+        target_type = self.get_target_type(data.data_type)
+        buf = site_buffer[target_type]
+
+        if data.data_type == AnalyticsDataType.PARAMETER:
+            buf.append(Param(data.tag, str(data.value)))
+        elif data.data_type == AnalyticsDataType.TAG:
+            buf.append(RunTag(data.tag, str(data.value)))
+        elif data.data_type == AnalyticsDataType.METRIC:
+            buf.append(Metric(data.tag, data.value, get_current_time_millis(), data.step or 0))
+        elif data.data_type == AnalyticsDataType.PARAMETERS:
+            for k, v in data.value.items():
+                buf.append(Param(k, str(v)))
+        elif data.data_type == AnalyticsDataType.TAGS:
+            for k, v in data.value.items():
+                buf.append(RunTag(k, str(v)))
+        elif data.data_type == AnalyticsDataType.METRICS:
+            for k, v in data.value.items():
+                buf.append(Metric(k, v, get_current_time_millis(), data.step or 0))
+
+    def get_target_type(self, data_type: AnalyticsDataType):
+        if data_type == AnalyticsDataType.METRIC:
+            return AnalyticsDataType.METRICS
+        elif data_type == AnalyticsDataType.PARAMETER:
+            return AnalyticsDataType.PARAMETERS
+        elif data_type == AnalyticsDataType.TAG:
+            return AnalyticsDataType.TAGS
+        else:
+            return data_type
+
+    def flush_buffer(self, record_origin, force_flush: bool = False):
```

Comment:
Suggest remove the "force_flush" option to be consistent with general "flush" method definition.

Suggets move the check for whether to flush or not logic in "save" method:


```
self.time_taken += timeit.default_timer() - self.time_start
if self.time_taken >= self.buff_flush_time:
    flush_buffer()
```